src/main/java/com/google/devtools/build/lib/sandbox/WindowsSandboxedSpawnRunner.java

```diff
@@ -34,62 +33,57 @@
 
   private final Path execRoot;
   private final PathFragment windowsSandbox;
-  private final Path sandboxBase;
   private final LocalEnvProvider localEnvProvider;
   private final Duration timeoutKillDelay;
-  private final TreeDeleter treeDeleter;
 
   /**
    * Creates a sandboxed spawn runner that uses the {@code windows-sandbox} tool.
    *
    * @param cmdEnv the command environment to use
-   * @param sandboxBase path to the sandbox base directory
    * @param timeoutKillDelay an additional grace period before killing timing out commands
+   * @param windowsSandboxPath path to windows-sandbox binary
    */
   WindowsSandboxedSpawnRunner(
       CommandEnvironment cmdEnv,
-      Path sandboxBase,
       Duration timeoutKillDelay,
-      PathFragment windowsSandboxPath,
-      TreeDeleter treeDeleter) {
+      PathFragment windowsSandboxPath) {
     super(cmdEnv);
     this.execRoot = cmdEnv.getExecRoot();
     this.windowsSandbox = windowsSandboxPath;
-    this.sandboxBase = sandboxBase;
     this.timeoutKillDelay = timeoutKillDelay;
     this.localEnvProvider = new WindowsLocalEnvProvider(cmdEnv.getClientEnv());
-    this.treeDeleter = treeDeleter;
   }
 
   @Override
   protected SpawnResult actuallyExec(Spawn spawn, SpawnExecutionContext context)
       throws IOException, ExecException, InterruptedException {
-    // Each invocation of "exec" gets its own sandbox base.
-    // Note that the value returned by context.getId() is only unique inside one given SpawnRunner,
-    // so we have to prefix our name to turn it into a globally unique value.
-    Path sandboxPath =
-        sandboxBase.getRelative(getName()).getRelative(Integer.toString(context.getId()));
-    sandboxPath.createDirectoryAndParents();
-
-    // b/64689608: The execroot of the sandboxed process must end with the workspace name, just like
-    // the normal execroot does.
-    Path sandboxExecRoot = sandboxPath.getRelative("execroot").getRelative(execRoot.getBaseName());
-    sandboxExecRoot.createDirectoryAndParents();
-
     Path tmpDir = createActionTemp(execRoot);
     Path commandTmpDir = tmpDir.getRelative("work");
     commandTmpDir.createDirectory();
     Map<String, String> environment =
         localEnvProvider.rewriteLocalEnv(
```

Comment:
Ideally `LocalEnvProvider.rewriteLocalEnv` would declare to return a `ImmutableMap` since all of its implementations actually do so.

Could you please change `environment` to `ImmutableMap` and use `ImmutableMap.copyOf(localEnvProvider.rewriteLocalEnv(...))` until that is fixed?